core/domain/state_domain.py

```diff
@@ -1925,6 +1928,341 @@ def validate(self) -> None:
                 self.duration_secs)
 
 
+def validate_rte_tags(
+    html: str, is_tags_nested_inside_tabs_or_collapsible: bool = False
+) -> None:
+    """Validate all the RTE tags.
+
+    Args:
+        html: str. The RTE content of the state.
+        is_tags_nested_inside_tabs_or_collapsible: bool. True when we
+            validate tags inside `Tabs` or `Collapsible` tag.
+
+    Raises:
+        utils.ValidationError. Raises the validation error in case the
+            content is not valid.
+    """
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+
+    for tag in soup.find_all('oppia-noninteractive-image'):
+        if not tag.has_attr('alt-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'alt-with-value\' attribute.'
+            )
+
```

Comment:
remove newlines between the if statements.

Reply:
Done